code/modules/multiz/zmimic/mimic_movable.dm

```diff
@@ -112,33 +122,37 @@
 		c_list[CL_MATRIX_AB] *= SHADOWER_DARKENING_FACTOR
 		color = c_list
 	else
-		// Not a color matrix, so we can just use the color var ourselves.
+		// Not a color matrix, so we just ignore the lighting values.
+		icon_state = "blank"	// this is actually just a white sprite, which is what this blending needs
 		color = list(
 			SHADOWER_DARKENING_FACTOR, 0, 0,
 			0, SHADOWER_DARKENING_FACTOR, 0,
 			0, 0, SHADOWER_DARKENING_FACTOR
 		)
 
-	if (our_overlays || priority_overlays)
-		compile_overlays()
-	else if (bound_overlay)
-		// compile_overlays() calls update_above().
+	var/turf/parent = loc
+	ASSERT(isturf(parent))
+	if (LAZYLEN(parent.ao_overlays_mimic))
+		overlays += parent.ao_overlays_mimic
```

Comment:
Don't manually manipulate overlays, the existing compile_overlays code you replaced was correct.